onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -707,7 +707,7 @@ class PlannerImpl {
 Status PlannerImpl::CreatePlan() {
   auto& p_graph_nodes = graph_viewer_.GetNodesInTopologicalOrder();
 
-  int num_ml_values = ort_value_name_idx_map_.MaxIdx() + 1;
+  int num_ml_values = ort_value_name_idx_map_.MaxIdx();
 
```

Comment:
Value was off by one, leading to one unused location slot. Broke tests when the default init of OrtMemoryInfo lacked clarity that it was invalid.

Reply:
all_values_size_ inside the IExecutionFrame constructor also adds the 1. The idea was that MaxIdx returned the maximum index stored in ort_value_name_idx_map_.

Reply:
I'll update OrtValueNameIdxMap::MaxIdx() to return what the name suggests.